### PR TITLE
fix violations on github/gitlab being struck-out all the time

### LINKED ISSUE #1189: Inline comments are always stuck out
# Report

## What did you do?

I am running Danger on Bitrise (using the [built-in step](https://www.bitrise.io/integrations/steps/danger)).

The command that is ran is:

    bundle "exec" "danger" "--fail-on-errors=true"

I configured the Dangerfile to have inline comments. It looks like the inline comments are always struck out. If I disable inline comments, the main comment sent to the PR has regular formatting.

At first, I thought that this was a bug in the `android_lint` plugin, but then I configured our iOS project with `swiftlint` and the behavior is the same.

## What did you expect to happen?

The inline comments show without a strike.

Here is an example without inline comments. The text is normal.

Screenshot 2020-02-19 at 07 39 32

## What happened instead?

Screenshot 2020-02-19 at 07 37 13

As you can see, the text is struck out, which is not expected. It looks like the error was corrected, which is not the case.

## Your Environment

* Which CI are you running on?
  * Bitrise
* Are you running the latest version of Danger?
  * Yes, 6.2.2
* What is your Dangerfile?

On iOS:

  ```ruby
  swiftlint.config_file = 'app/.swiftlint.yml'
  swiftlint.verbose = true
  swiftlint.max_num_violations = 20
  swiftlint.lint_files inline_mode: true
  ```
On Android:

  ```ruby
  checkstyle_dir = "**/build/reports/detekt/detekt.xml"
  Dir[checkstyle_dir].each do |file_name|
    checkstyle_format.base_path = Dir.pwd
    checkstyle_format.report file_name
  end

  lint_dir = "**/build/reports/lint-**.xml"
  Dir[lint_dir].each do |file_name|
    android_lint.skip_gradle_task = true
    android_lint.filtering = true
    android_lint.report_file = file_name
    android_lint.lint(inline_mode: true)
  end

  junit_tests_dir = "**/test-results/**/*.xml"
  Dir[junit_tests_dir].each do |file_name|
    junit.parse file_name
    junit.report
  end
  ```

### PR DESCRIPTION
**Pull request by telyn**
Hopefully fixes #1189.

I'd like confirmation that this fix works - I don't at the moment have any projects using github/gitlab as the request source.

### CODE REVIEW

**Comment by danger-public** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry. <br />
You can find it at <a href="https://github.com/danger/danger/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Tests were not updated</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by telyn** [comment]
Cancel that - this is definitely not the correct way to fix this problem - I see that now!

**Comment by orta** [comment]
If you're happy - this looks good to me

**Comment by telyn** [comment]
I'm happy that it's the right way to fix it - not totally convinced I *have* fixed it, if that makes sense. Just want someone to say "Works for me!" and then I'll be happy.

No sense in merging an untested fix!

**Comment by orta** [comment]
language without types y'know - can you _really_ know anything? We can ask in that thread, or you could add `warn("Test", file: "CHANGELOG.md", line: 10)` to the dangerfile in here and see if it works?

**Comment by telyn** [comment]
> language without types y'know - can you _really_ know anything? We can ask in that thread, or you could add `warn("Test", file: "CHANGELOG.md", line: 10)` to the dangerfile in here and see if it works?

I'm a bit of a doofus for not thinking of that. Here goes!

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,11 +10,14 @@
```
Comment (danger-public) [inline CHANGELOG.md]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/danger/danger/blob/ebf87d246e5948a50c1d85c6dae3022c1ca2d02e/CHANGELOG.md#L10"></span>Test</td>
    </tr>
  </tbody>
</table>

Comment (telyn) [reply CHANGELOG.md]: woop!

**Comment by telyn** [comment]
cleaner to remove that little test from history :)

**Comment by orta** [comment]
Perfect, lets get this shipped

**Comment by orta** [comment]
Deployed

**Comment by uziasferreirazup** [comment]
I hae the same issue, in the version 7.0.1 :(

**Comment by uziasferreirazup** [comment]
I don't what happened, but I search the change in the master and the change not existing

**Pull request merged**